### ISSUE TITLE
Sometimes DittoClients.newInstance(...) blocks forever

### ISSUE THREAD

**Issue by desislava-marinova** [eclipse/ditto-clients#89]
Creating ditto client sometimes blocks forever. Here is the thread that is blocked and never ends. 
```
"ditto-client--adaptable-bus-8335710b-34bb-4263-8294-207f2e63bfcc-2" #396 prio=5 os_prio=0 tid=0x000000001f745000 nid=0x338c waiting on condition [0x000000002b6ec000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000000e2bfc658> (a java.util.concurrent.CompletableFuture$Signaller)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
        at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
        at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
        at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1934)
        at org.eclipse.ditto.client.messaging.internal.WebSocketMessagingProvider.initialize(WebSocketMessagingProvider.java:171)
        - locked <0x00000000e2c0c448> (a java.util.concurrent.atomic.AtomicBoolean)
        - locked <0x00000000e2c0b990> (a org.eclipse.ditto.client.messaging.internal.WebSocketMessagingProvider)
        at org.eclipse.ditto.client.internal.DefaultDittoClient.init(DefaultDittoClient.java:240)
        at org.eclipse.ditto.client.internal.DefaultDittoClient.configureTwin(DefaultDittoClient.java:197)
        at org.eclipse.ditto.client.internal.DefaultDittoClient.newInstance(DefaultDittoClient.java:123)
        at org.eclipse.ditto.client.DittoClients.newInstance(DittoClients.java:132)
        at org.eclipse.ditto.client.DittoClients.newInstance(DittoClients.java:90)
        at org.eclipse.ditto.client.DittoClients.newInstance(DittoClients.java:68)
        at org.eclipse.ditto.client.DittoClients.newInstance(DittoClients.java:51)
        at com.bosch.iot.dm.things.access.DittoClientFactory.lambda$createClient$9(DittoClientFactory.java:305)
        at com.bosch.iot.dm.things.access.DittoClientFactory$$Lambda$799/26418538.call(Unknown Source)
        at com.bosch.iot.dm.vertx.Utils.outOfEventLoop(Utils.java:410)
        at com.bosch.iot.dm.vertx.Utils.outOfEventLoop(Utils.java:368)
        at com.bosch.iot.dm.things.access.DittoClientFactory.createClient(DittoClientFactory.java:305)
        at com.bosch.iot.dm.things.access.DittoClientFactory.getInstance(DittoClientFactory.java:286)
        - locked <0x00000000e169b460> (a java.util.HashMap)
        at com.bosch.iot.dm.things.access.DittoClientFactory$$Lambda$1359/687663372.apply(Unknown Source)
        at com.bosch.iot.dm.async.CompletableFuture.lambda$null$24(CompletableFuture.java:982)
        at com.bosch.iot.dm.async.CompletableFuture$$Lambda$778/1637673631.call(Unknown Source)
        at com.bosch.iot.dm.auth.AuthCtx.callWithMDCSubscrId(AuthCtx.java:300)
        at com.bosch.iot.dm.auth.AuthCtx.callWithMDCAuthCtx(AuthCtx.java:280)
        at com.bosch.iot.dm.auth.AuthCtx.callAs(AuthCtx.java:239)
        at com.bosch.iot.dm.auth.AuthCtx.callAs(AuthCtx.java:216)
        at com.bosch.iot.dm.async.CompletableFuture.lambda$authCtxAware$25(CompletableFuture.java:982)
        at com.bosch.iot.dm.async.CompletableFuture$$Lambda$777/2135287206.apply(Unknown Source)
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
        at com.bosch.iot.dm.async.CompletableFuture.thenCompose(CompletableFuture.java:758)
        at com.bosch.iot.dm.async.CompletableFuture.thenCompose(CompletableFuture.java:61)
        at com.bosch.iot.dm.things.access.DittoClientFactory.apply(DittoClientFactory.java:230)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.lambda$definitionMatch$45(FeatureEventSource.java:256)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource$$Lambda$2075/234707026.call(Unknown Source)
        at com.bosch.iot.dm.auth.AuthCtx.callWithMDCSubscrId(AuthCtx.java:304)
        at com.bosch.iot.dm.auth.AuthCtx.callWithMDCAuthCtx(AuthCtx.java:284)
        at com.bosch.iot.dm.auth.AuthCtx.callAs(AuthCtx.java:243)
        at com.bosch.iot.dm.auth.AuthCtx.callAs(AuthCtx.java:216)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.definitionMatch(FeatureEventSource.java:256)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.lambda$null$33(FeatureEventSource.java:242)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource$$Lambda$2074/1300308500.apply(Unknown Source)
        at java.util.Optional.map(Optional.java:215)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.lambda$definitionMatch$35(FeatureEventSource.java:242)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource$$Lambda$2073/1013156636.apply(Unknown Source)
        at java.util.Optional.map(Optional.java:215)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.definitionMatch(FeatureEventSource.java:241)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.featureChangeMatch(FeatureEventSource.java:233)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.lambda$match$27(FeatureEventSource.java:221)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource$$Lambda$2064/1538987199.apply(Unknown Source)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
        at com.bosch.iot.dm.devices.events.impl.FeatureEventSource.match(FeatureEventSource.java:222)
        at com.bosch.iot.dm.devices.events.impl.DeviceStatusEventSource.lambda$match$11(DeviceStatusEventSource.java:163)
        at com.bosch.iot.dm.devices.events.impl.DeviceStatusEventSource$$Lambda$2056/1116002950.apply(Unknown Source)
        at java.util.Optional.map(Optional.java:215)
        at com.bosch.iot.dm.devices.events.impl.DeviceStatusEventSource.match(DeviceStatusEventSource.java:160)
        at com.bosch.iot.dm.devices.events.impl.DeviceStatusEventSource.lambda$subscribe$17(DeviceStatusEventSource.java:182)
        at com.bosch.iot.dm.devices.events.impl.DeviceStatusEventSource$$Lambda$805/1003587043.accept(Unknown Source)
        at org.eclipse.ditto.client.internal.bus.SelectorUtil.lambda$registerForChanges$9(SelectorUtil.java:236)
        at org.eclipse.ditto.client.internal.bus.SelectorUtil$$Lambda$1284/2117200343.accept(Unknown Source)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus.lambda$notify$1(DefaultPointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus$$Lambda$1493/1048547345.accept(Unknown Source)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus.notify(DefaultPointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.PointerBus.notify(PointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.SelectorUtil.lambda$addHandlerForThingEvent$4(SelectorUtil.java:149)
        at org.eclipse.ditto.client.internal.bus.SelectorUtil$$Lambda$1032/588256040.accept(Unknown Source)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus.lambda$notify$1(DefaultPointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus$$Lambda$1493/1048547345.accept(Unknown Source)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
        at org.eclipse.ditto.client.internal.bus.DefaultPointerBus.notify(DefaultPointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.PointerBus.notify(PointerBus.java:49)
        at org.eclipse.ditto.client.internal.bus.PointerBus.notify(PointerBus.java:37)
        at org.eclipse.ditto.client.internal.CommonManagementImpl.lambda$null$17(CommonManagementImpl.java:630)
        at org.eclipse.ditto.client.internal.CommonManagementImpl$$Lambda$1450/170622711.accept(Unknown Source)
        at org.eclipse.ditto.client.internal.CommonManagementImpl.lambda$subscribeAndPublishMessage$19(CommonManagementImpl.java:650)
        at org.eclipse.ditto.client.internal.CommonManagementImpl$$Lambda$1292/680998013.accept(Unknown Source)
        at org.eclipse.ditto.client.internal.bus.DefaultAdaptableBus.lambda$runConsumerAsync$6(DefaultAdaptableBus.java:214)
        at org.eclipse.ditto.client.internal.bus.DefaultAdaptableBus$$Lambda$1340/2080062198.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by DerSchwilk** [comment]
This could potentially be caused by #88 and fixed with the corresponding PR.

**Comment by thjaeckle** [comment]
Also fixed by #88
